README.md

```diff
@@ -2,8 +2,120 @@
+   ```
+  - bash
+   ```
+   echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ~/.bash_profile
+   ```
+  - zsh
+   ```
+   echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ${ZDOTDIR:-~}/.zshrc
+   ```
+3. install direnv plugin
+```
+asdf plugin-add direnv
+asdf install direnv latest
+asdf global direnv latest
+```
+4. integrate direnv plugin with your shell
+  - fish
+   ```
+   printf "\
+   # Hook direnv into your shell.
+   asdf exec direnv hook fish | source
+
+   # A shortcut for asdf managed direnv.
+   function direnv
+     asdf exec direnv \"\$argv\"
+   end
+   " >> ~/.config/fish/config.fish
+   ```
+  - bash
+   ```
+   cat << EOF >> ~/.bashrc
+   # Hook direnv into your shell.
+   eval "$(asdf exec direnv hook bash)"
+
+   # A shortcut for asdf managed direnv.
+   direnv() { asdf exec direnv "$@"; }
+   EOF
+   ```
+  - zsh
+   ```
+   cat << EOF >>  ~/.zshrc
+   # Hook direnv into your shell.
+   eval "$(asdf exec direnv hook zsh)"
+
+   # A shortcut for asdf managed direnv.
+   direnv() { asdf exec direnv "$@"; }
+   EOF
+   ```
+5. enable `use asdf` feature
+```
+echo 'source "$(asdf direnv hook asdf)"' >> ~/.config/direnv/direnvrc
+```
+6. exit from the current shell or start a new one
+7. `cd` into the directory where you checked out `clouseau` and do `direnv allow`
+8. do `make tools` (we need it only when we add new tools into `.tool-versions`)
+
+Now every time you `cd` into the project directory the correct versions of tools
+will be activated.
+
+### Custom Java
+
+In some cases version of Java is not available in asdf-java.
+There is a trick which makes globally available version recognized by
+asdf.
+
+```
+ln -s /Library/Java/JavaVirtualMachines/jdk1.7.0_80.jdk/Contents/Home ~/.asdf/installs/java/oracle-1.7.0
+asdf reshim java
+```
+
+## Building distribution
+
+Just run `make` to get `target/clouseau-{version}-SNAPSHOT.zip` file.
+
+## Running locally
+
+To run single node instance use `make clouseau1`.
+
+Usually for testing all three nodes might be needed. In such case run following
+from separate terminals:
+
+```
+make clouseau1
+make clouseau2
+make clouseau3
+```
+
+## Changing tools version
+
+The list of tools are configured in a `.tool-versions` file. Make sure you
+run `make tools` every time you add new tools. You don't need to do it when
+there is just a version change.
+
 ## Configuration options
-This guide explains the various clouseau configuration options available, and how to use them to tune clouseau performance and scalability. There are two categories of clouseau options, first category is about tuning the JVM (ex: Xmx) and other category of options that go into clouseau.ini. 
+This guide explains the various clouseau configuration options available, and how to use them to tune clouseau performance and scalability. There are two categories of clouseau options, first category is about tuning the JVM (ex: Xmx) and other category of options that go into clouseau.ini.
```

Comment:
I am sorry my editor removes trailing spaces. I can "fix it" if it is an issue.

Reply:
This is a good thing, I do not think we should worry about that.